### ISSUE TITLE
Cannot send email and share with collaborator.

### ISSUE THREAD

**Issue by EadrenKing** [sharelatex/sharelatex#577]
## Problem

Cannot send email and share with collaborator.

## Install type

using the Docker image (as described at https://github.com/sharelatex/sharelatex/wiki/Quick-Start-Guide)

## Version

latest

### /var/log/sharelatex/web.log

```
{"name":"web-sharelatex","hostname":"5538749ca745","pid":112,"level":50,"email"::
"jinyejun@ncic.ac.cn","msg":"error sending test email","time":"2018-01-13T16:38::
51.921Z","v":0}
{"name":"web-sharelatex","hostname":"5538749ca745","pid":112,"level":50,"err":{""
message":"Cannot send email","name":"Error","stack":"Error: Cannot send email\n
 at /var/www/sharelatex/web/app/js/Features/Email/EmailSender.js:126:19\n  at Noo
demailer.<anonymous> (/var/www/sharelatex/web/node_modules/nodemailer/lib/nodemaa
iler.js:213:26)\n  at SMTPConnection.<anonymous> (/var/www/sharelatex/web/node_mm
odules/nodemailer-smtp-transport/lib/smtp-transport.js:80:16)\n  at SMTPConnectii
on.g (events.js:292:16)\n  at emitOne (events.js:96:13)\n  at SMTPConnection.emii
t (events.js:188:7)\n  at SMTPConnection._onError (/var/www/sharelatex/web/node__
modules/smtp-connection/lib/smtp-connection.js:414:10)\n  at emitOne (events.js::
96:13)\n  at TLSSocket.emit (events.js:188:7)\n  at emitErrorNT (net.js:1277:8)\\
n  at _combinedTickCallback (internal/process/next_tick.js:80:11)\n  at process..
_tickDomainCallback (internal/process/next_tick.js:128:9)\n"},"url":"/launchpad//
send_test_email","method":"POST","user":{"id":"5a5a35e5435c9d007081665f","email""
:"jinyejun@ncic.ac.cn"},"msg":"error passed to top level next middlewear","time""
:"2018-01-13T16:38:51.921Z","v":0}
```

## docker-compose.yml

version: '2'
services:
    sharelatex:
        restart: always
        image: sharelatex/sharelatex
        container_name: sharelatex
        depends_on:
            - mongo
            - redis
        privileged: true
        ports:
            - 80:80 
        links:
            - mongo
            - redis
        volumes:
            - ./sharelatex_data:/var/lib/sharelatex
        
        environment:
            SHARELATEX_MONGO_URL: mongodb://mongo/sharelatex
            SHARELATEX_REDIS_HOST: redis
            SHARELATEX_APP_NAME: Our ShareLaTeX
            
            #Set for SSL via nginx-proxy
            #VIRTUAL_HOST: 103.112.212.22

            # SHARELATEX_SITE_URL: http://sharelatex.mydomain.com
            # SHARELATEX_NAV_TITLE: Our ShareLaTeX Instance
            # SHARELATEX_HEADER_IMAGE_URL: http://somewhere.com/mylogo.png
            # SHARELATEX_ADMIN_EMAIL: support@it.com

            # SHARELATEX_LEFT_FOOTER: '[{"text": "Powered by <a href=\"https://www.sharelatex.com\">ShareLaTeX</a> 2016"},{"text": "Another page I want to link to can be found <a href=\"here\">here</a>"} ]'
            # SHARELATEX_RIGHT_FOOTER: '[{"text": "Hello I am on the Right"} ]'

            SHARELATEX_EMAIL_FROM_ADDRESS: "jinyeju@ncic.ac.cn" # my center email

            # SHARELATEX_EMAIL_AWS_SES_ACCESS_KEY_ID: 
            # SHARELATEX_EMAIL_AWS_SES_SECRET_KEY: 

            SHARELATEX_EMAIL_SMTP_HOST: "smtp.ncic.ac.cn"
            SHARELATEX_EMAIL_SMTP_PORT: 25 
            SHARELATEX_EMAIL_SMTP_SECURE: "false"
            SHARELATEX_EMAIL_SMTP_USER: "jinyejun@ncic.ac.cn"
            SHARELATEX_EMAIL_SMTP_PASS: "mypassword"
            SHARELATEX_EMAIL_SMTP_TLS_REJECT_UNAUTH: "true"
            SHARELATEX_EMAIL_SMTP_IGNORE_TLS: "false"
            SHARELATEX_CUSTOM_EMAIL_FOOTER: "<div>This system is run by HPC </div>"

            ################
            ## Server Pro ##
            ################

            # SANDBOXED_COMPILES:true

            # SHARELATEX_LDAP_HOST: 'ldap://ldap.forumsys.com' 
            # SHARELATEX_LDAP_DN: 'uid=:userKey,dc=example,dc=com' 
            # SHARELATEX_LDAP_BASE_SEARCH: 'dc=example,dc=com'
            # SHARELATEX_LDAP_FILTER: '(uid=:userKey)'
            # SHARELATEX_LDAP_ADMIN_DN: 'cn=read-only-admin,dc=example,dc=com' 
            # SHARELATEX_LDAP_ADMIN_PW: password'
            # SHARELATEX_LDAP_ANONYMOUS: "false"
            # SHARELATEX_LDAP_EMAIL_ATT: "mail"
            # SHARELATEX_LDAP_NAME_ATT: "name"
            # SHARELATEX_LDAP_LAST_NAME_ATT: "secondname"
            # SHARELATEX_LDAP_PLACEHOLDER: "Username"
            # SHARELATEX_LDAP_TLS: "true"
            # SHARELATEX_LDAP_TLS_OPTS_REJECT_UNAUTH:
            # SHARELATEX_LDAP_TLS_OPTS_CA_PATH: '["/var/one.pem", "/var/two.pem"]'

            # SHARELATEX_TEMPLATES_USER_ID: "578773160210479700917ee5"

            # SHARELATEX_PROXY_LEARN: "true"

    mongo:
        restart: always
        image: mongo
        container_name: mongo
        expose:
            - 27017
        volumes:
            - ./mongo_data:/data/db

    redis:
        restart: always
        image: redis
        container_name: redis
        expose:
            - 6379
        volumes:
            - ./redis_data:/data

    # nginx-proxy:
      # image: jwilder/nginx-proxy
      # container_name: nginx-proxy
      # ports:
        #- "80:80"
        # - "443:443"
        # volumes:
        # - ./var/run/docker.sock:/tmp/docker.sock:ro
        # - ./sharelatex/tmp:/etc/nginx/certs

## Screenshots

￼
![wechatimg21](https://user-images.githubusercontent.com/25998993/34908066-a64921a2-f8c4-11e7-8771-1bd646c98253.jpeg)

**Comment by jdleesmiller** [comment]
Thanks for getting in touch and for providing a detailed report. I'm afraid I don't see anything wrong with the config. The error message isn't very informative, unfortunately. Looking at where it comes from:

https://github.com/sharelatex/web-sharelatex/blob/75db227732361589f110adafeab37ac968dc11c1/app/coffee/Features/Email/EmailSender.coffee#L73-L78

It looks like the original error should be logged to the web error logs. Maybe you can find the original error message in those logs?

Since there's already an issue on the docker image, I'll close this one --- if you find more info in the logs, perhaps you can post it on the other issue.

**Comment by EadrenKing** [comment]
But could you provide a detail example of config docker compose file for me.

**Comment by EadrenKing** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
But could you provide a detail example of config docker compose file for me.



    
        a#ntes-pcmail-signature-default:hover {
            text-decoration: underline;
            color: #3593db;
            cursor: pointer;
        }
    




    On 01/15/2018 19:29，John Lees-Miller<notifications@github.com> wrote： 


Closed #577.
